### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by jglick**
Clone of https://github.com/jenkinsci/maven-hpi-plugin/pull/156.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,28 +1,11 @@
...
-                    }
-                }
+properties([buildDiscarder(logRotator(numToKeepStr: '20'))])
+node('maven') {
+    checkout scm
+    timeout(time: 1, unit: 'HOURS') {
+        // TODO Azure mirror
+        ansiColor('xterm') {
+            withEnv(['MAVEN_OPTS=-Djansi.force=true']) {
+                sh 'mvn -B -Dstyle.color=always -ntp -Prun-plugin-pom-its clean verify'
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Could we use `runMaven` instead? Mirros & Co

Comment (jglick) [reply Jenkinsfile]: `runMaven` does not set up mirrors, nor has it been updated for `-ntp`, and does a bunch of stuff which is useless in an ACI agent. I prefer to keep things more transparent and explicit.

Comment (oleg-nenashev) [reply Jenkinsfile]: > runMaven does not set up mirrors, nor has it been updated for -ntp, and does a bunch of stuff which is useless in an ACI agent

Hm? https://github.com/jenkins-infra/pipeline-library/blob/master/vars/infra.groovy#L69-L89

Maybe you are talking about the Pipeline Maven Plugin. Sorry, I meant `infra.runMaven`. We can add `-ntp` there and any ACI agent tweaks if needed there

Comment (jglick) [reply Jenkinsfile]: Can be done in a follow-up, though I would still prefer to call `retrieveMavenSettingsFile` directly and avoid `runMaven`: nicer to have a single command you can copy and paste into a local shell to check build failures. Excessive Groovy functions just obscures what is actually being run.

**Pull request merged**